### PR TITLE
Validation improvment (filters and followers table)

### PR DESCRIPTION
**Pull request by SebastienPeillet**
Add filter on validation button (pen button). Two filters can be added, to only show the day entities and/or the unvalidated entities. Futhermore I also add a option in follower button to open the followar table in order to validate followers. The followers table also uses the benefit of the two filters described above.

### CODE REVIEW

**Comment by SebastienPeillet** [comment]
@pblottiere It's rebased you can test it

**Comment by pblottiere** [comment]
I don't think it's due to this MR but when I click on the `Validate (all or current selection)`, I have the next error:

```
Traceback (most recent call last):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/sammo.py", line 289, in validate
    self.session.validate()
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/session.py", line 366, in validate
    next(self.surveyLayer.getFeatures())
StopIteration
```

**Comment by pblottiere** [comment]
> I don't think it's due to this MR but when I click on the Validate (all or current selection), I have the next error:

And if I add a survey, it's complaining about the transect (and so on). Should we manage the fact that settings table could be empty?

**Comment by pblottiere** [comment]
Weirdly the `species` column of the first row is focused/selected when I activate/deactivate the 2 new filters:

image

**Comment by pblottiere** [comment]
But otherwise, everything works well :+1:

File: sammo.py [hunk 1]
```diff
@@ -419,17 +444,24 @@ def onEnvironmentAction(self) -> None:
...
     def onSightingsAction(self):
         self.soundRecordingController.onStartSightings()
         self.iface.mapCanvas().setFocus()
         layer = self.session.addSightingsFeature()
-        self.tableDock.refresh(layer)
+        self.tableDock.refresh(layer, self.filterExpr)
         self.soundRecordingController.onStopEventWhichNeedSoundRecord(60)
 
-    def onFollowersAction(self):
+    def onFollowersAction(self, validation: QAction):
```
Comment (pblottiere) [inline sammo.py:457]:
```suggestion
    def onFollowersAction(self, validation: QAction) -> None:
```

File: src/gui/attribute_table.py [hunk 2]
```diff
@@ -15,9 +21,10 @@ def toolbar(table):
         return table.findChild(QToolBar, "mToolbar")
 
     @staticmethod
-    def refresh(table, layerName):
-        table.findChild(QAction, "mActionReload").trigger()
+    def refresh(table, layerName, filterExpr="True"):
```
Comment (pblottiere) [inline src/gui/attribute_table.py:24]: can we add the typing while being on this method?

File: src/gui/table.py [hunk 3]
```diff
@@ -107,9 +107,9 @@ def init(self, environmentLayer, sightingsLayer):
     def unload(self):
         self.clean()
 
-    def refresh(self, layer):
+    def refresh(self, layer, filterExpr="True"):
```
Comment (pblottiere) [inline src/gui/table.py:110]: typing

Comment (SebastienPeillet) [reply src/gui/table.py]: Done !

File: src/gui/attribute_table.py (hunk 2 above)
Comment (SebastienPeillet) [reply src/gui/attribute_table.py]: done !

**Comment by SebastienPeillet** [comment]
> And if I add a survey, it's complaining about the transect (and so on). Should we manage the fact that settings table could be empty?

It's the same error you already add on the QGIS 3.26 version, when a empty featureCount() equals -1 and not 0.

**Comment by SebastienPeillet** [comment]
> Weirdly the species column of the first row is focused/selected when I activate/deactivate the 2 new filters:

It's because we refreshing the table, as when we add a new line. I fixed it

**Comment by pblottiere** [comment]
>>    And if I add a survey, it's complaining about the transect (and so on). Should we manage the fact that settings table could be empty?

> It's the same error you already had on the QGIS 3.26 version, when a empty featureCount() equals -1 and not 0.

Argh ok.

**Comment by pblottiere** [comment]
Everything looks good thanks. +1 to merge when you're ready.

**Pull request merged**